Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -341,7 +341,17 @@ extension RawSyntax {
   }
 }
 
-extension RawSyntax: Codable {
+// This is needed purely to have a self assignment initializer for
+// RawSyntax.init(from:Decoder) so we can reuse omitted nodes, instead of
+// copying them.
+protocol _RawSyntaxFactory {}
```

Comment:
Sorry, this can be `private`, and we don't need `fileprivate` in `init(_inistance)` decl.